packages/drivers/local-driver/src/localDocumentDeltaConnection.ts

```diff
@@ -62,24 +62,31 @@ export class LocalDocumentDeltaConnection extends DocumentDeltaConnection {
           super(socket, documentId, new TelemetryNullLogger());
     }
 
+    protected submitCore(type: string, messages: IDocumentMessage[]) {
+        if (this.isBatchManagerDisabled) {
+            this.emitMessages(type, [messages]);
+        } else {
+            this.submitManager.add(type, messages);
+            this.submitManager.drain();
+        }
+    }
+
     /**
      * Submits a new delta operation to the server
      */
     public submit(messages: IDocumentMessage[]): void {
         // We use a promise resolve to force a turn break given message processing is sync
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         Promise.resolve().then(() => {
-            this.submitManager.add("submitOp", messages);
-            this.submitManager.drain();
+            this.submitCore("submitOp", messages);
```

Comment:
[messages] ?
Just to make sure that contract is clear: it's an array of batches, where batch can be either single op or many ops. So submitting just an array is valid, but it breaks contract - it no longer a single batch.
And while it's valid RE protocol, I think we should disallow sending array of ops. It should be array of arrays always, where inner array can have just one op. That way we should be able to have proper type guards in this code and catch cases like this where we are passing just an array.

Another solution - stop supporting multiple batches in these layers (as input). I.e. multiple batches being sent at once can only happen due to BatchManager, so it feels like all APIs before should just take an array (single batch), and there should be no need to have [messages[ in other places

Reply:
Curious, is there a way to write UT that will validate that messages on socket show up in right form (i.e. batch is preserved)?

Reply:
Per other comment, this seems correct, but question about UT might be still relevant (though TSC properly validates types, then it's less of an issue)

Reply:
The submitCore implementation takes care of that. It has the same exact signature as `batchmanager.add`. There's no regression here.